### ISSUE TITLE
go/heif does not compile and tests fail with v1.18.2

### ISSUE THREAD

**Issue by mholt** [strukturag/libheif#1264]
Under v1.18.2, the tests fail:

```
╔[matt@shadowfax:~/go/pkg/mod/github.com/strukturag/libheif@v1.18.2/go/heif]
╚>$ go test *.go
# command-line-arguments
# [command-line-arguments]
./heif.go:58:28: could not determine kind of name for C.heif_compression_HTJ2K
./heif.go:270:43: could not determine kind of name for C.heif_suberror_Camera_extrinsic_matrix_undefined
./heif.go:268:43: could not determine kind of name for C.heif_suberror_Camera_intrinsic_matrix_undefined
./heif.go:281:35: could not determine kind of name for C.heif_suberror_Compression_initialisation_error
./heif.go:272:37: could not determine kind of name for C.heif_suberror_Decompression_invalid_data
./heif.go:324:33: could not determine kind of name for C.heif_suberror_Invalid_J2K_codestream
./heif.go:328:22: could not determine kind of name for C.heif_suberror_No_icbr_box
./heif.go:314:27: could not determine kind of name for C.heif_suberror_No_ispe_property
./heif.go:254:39: could not determine kind of name for C.heif_suberror_No_matching_decoder_installed
./heif.go:326:21: could not determine kind of name for C.heif_suberror_No_vvcC_box
./heif.go:340:48: could not determine kind of name for C.heif_suberror_Unsupported_generic_compression_method
FAIL	command-line-arguments [build failed]
FAIL
```

and my own program that uses this package also fails to compile after upgrading from v.17.6. Downgrading back to v1.17.6, the tests pass and my program compiles again:

```
╔[matt@shadowfax:~/go/pkg/mod/github.com/strukturag/libheif@v1.17.6/go/heif]
╚>$ go test *.go
ok  	command-line-arguments	1.958s
```

**Comment by bradh** [comment]
Do you have 1.17.6 installed? I have no idea about golang, but maybe its finding the installed `heif.h` instead of the one in the build tree.

**Comment by mholt** [comment]
You know, I thought I did, since running `pacman -Syu` said "up to date" and https://archlinux.org/packages/extra/x86_64/libheif/ showed a 1.18.* version, but using `pacman -Qi libheif` showed 1.17.6. Turns out my pacman mirror was... _checks notes_ ... 900+ hours out-of-date. For some reason I guess it stopped syncing? Anyway, I have never experienced that before (nor have I fiddled with mirrors before) so I changed to a mirror that updated a few seconds ago and now I have 1.18.2 installed on my system.

And now it works fine.

So... my apologies for the false alarm.
